packages/next/src/hooks/useAuth.ts

```diff
@@ -16,7 +27,8 @@ export function create<
     };
   } = never,
 >(): NextClient<Schema, ObjectTypesNames, ObjectTypes>['auth']['useAuth'] {
-  return () => {
+  return (useAuthOptions?: UseAuthOptions) => {
+    const { shouldRedirect = true } = useAuthOptions || {};
```

Comment:
We should use `defaults` from `lodash` here instead of `useAuthOptions || {}`. While it should work this way, it could have unintended consequences.